api/METRICS.md

```diff
@@ -0,0 +1,91 @@
+# API proposal for metrics
+
+## Example
+
+```python
+# For most sklearn metrics, we would have their group version that returns a Bunch with fields
+#   * overall: overall metric value
+#   * by_group: a dictionary that maps sensitive feature values to metric values
+
+summary = accuracy_score_by_group(y_true, y_pred, sensitive_features=sf, **other_kwargs)

```

Comment:
So, basically delete all the 'helper' fields from the `GroupMetricResult` and turn it into a `Bunch`. If we want to eliminate types where possible, this seems reasonable.